oak/server/storage/storage_processor.cc

```diff
@@ -34,26 +36,23 @@ std::string GetStorageId(const std::string& storage_name) {
   return storage_name;
 }
 
-asylo::CleansingVector<uint8_t> GetStorageEncryptionKey(const std::string& storage_name) {
-  // TODO: Request encryption key from escrow service.
-  std::string encryption_key =
-      absl::HexStringToBytes("c0dedeadc0dedeadc0dedeadc0dedeadc0dedeadc0dedeadc0dedeadc0dedead");
-  return asylo::CleansingVector<uint8_t>(encryption_key.begin(), encryption_key.end());
-}
-
 }  // namespace
 
-StorageProcessor::StorageProcessor(const std::string& storage_address)
+StorageProcessor::StorageProcessor(const std::string& storage_address,
+                                   StorageEncryptionKeyFunction storage_encryption_key_function)
     : fixed_nonce_generator_(new oak::FixedNonceGenerator()),
       datum_name_cryptor_(kMaxMessageSize, fixed_nonce_generator_),
       datum_value_cryptor_(kMaxMessageSize, new asylo::AesGcmSivNonceGenerator()),
       storage_service_(oak::Storage::NewStub(
-          grpc::CreateChannel(storage_address, grpc::InsecureChannelCredentials()))) {}
+          grpc::CreateChannel(storage_address, grpc::InsecureChannelCredentials()))),
+      storage_encryption_key_function_(storage_encryption_key_function) {}
 
 const asylo::StatusOr<std::string> StorageProcessor::EncryptDatum(const std::string& datum,
                                                                   DatumType datum_type) {
   // TODO: Replace "foo" with identifier for the encryption key.
-  asylo::CleansingVector<uint8_t> key = GetStorageEncryptionKey("foo");
+  asylo::CleansingVector<uint8_t> key;
+  ASYLO_ASSIGN_OR_RETURN(key, storage_encryption_key_function_("foo"));
```

Comment:
I am not sure I see what is doing the encryption / decryption of the escrowed secrets, and with what keys, and under what conditions these are released, etc. It's perfectly fine if these functions are not implemented yet, but could you add placeholders and TODOs to make sure they are tracked? Also it would be good to have this detailed (and discussed and reviewed) in the design doc as well.